wp-job-manager-autoload.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Autoload plugin classes.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Autoload plugin classes.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Autoload {
+
+	/**
+	 * Namespace -> directory mappings.
+	 *
+	 * @var array
+	 */
+	private static $autoload_map = [];
+
+	/**
+	 * Add the autoloader.
+	 */
+	public static function init() {
+		spl_autoload_register( [ self::class, 'autoload' ] );
+	}
+
+	/**
+	 * Register a new plugin with a class prefix and directory to autoload.
+	 *
+	 * @param string $namespace Root namespace. Should start with WP_Job_Manager_.
+	 * @param string $dir Directory to autoload.
+	 */
+	public static function register( $namespace, $dir ) {
+		self::$autoload_map[ $namespace ] = $dir;
+	}
+
+	/**
+	 * Autoload plugin classes.
+	 *
+	 * @access private
+	 *
+	 * @param string $class_name Class name.
+	 */
+	public static function autoload( $class_name ) {
+
+		if ( ! str_starts_with( $class_name, 'WP_Job_Manager' ) || ! str_contains( $class_name, '\\' ) ) {
+			return;
+		}
+
+		list( $namespace, $file_name ) = explode( '\\', $class_name, 2 );
+
+		$root_dir = self::$autoload_map[ $namespace ];
```

Comment:
Maybe it is worth checking if that `$namespace` in fact exists?

Otherwise, if someone tries to load a class from a `WP_Job_Manager` namespace and that namespace for some reason doesn't exist, we will have an "Undefined array key" error.

Reply:
For instance, while testing this PR I had the following notice emitted with the WPJM Alerts addon enabled:

```
Notice: Undefined index: WP_Job_Manager_Alerts in /var/www/html/wp-content/plugins/WP-Job-Manager/wp-job-manager-autoload.php on line 58
```

Reply:
Thanks, added some checks in 4ff906f 
Also removed some includes in 0b52eaa to start using this autoloader for the recent namespaced classes.

Reply:
Perfect! Thanks!